core/App/screens/Connection.tsx

```diff
@@ -60,6 +72,14 @@ const Connection: React.FC<ConnectionProps> = ({ navigation, route }) => {
     },
   })
 
+  useEffect(() => {
+    // FIX:(jl) There may be a better way to fetch queued messages.
+    // Under investigation.
+    if (connection && connection.state === 'completed') {
```

Comment:
Wondering if we should also do this if the state is in `request-sent` as well? In both those scenarios, ACA-Py considers the connection established.

Otherwise this LGTM as a workaround.

Reply:
Do you mean `response-sent`? If so I think completed is better because it is the eventual end state the connection gets to in AFJ. If we made the call during `response-sent` it would be called 2x almost back-to-back.